From d7577e1e55595123e3bcec78fa4a79fe8a314fe5 Mon Sep 17 00:00:00 2001
From: Li xin <lixin.fnst@cn.fujitsu.com>
Date: Sun, 19 Jul 2015 02:42:58 +0900
Subject: [PATCH] audit: Fixed swig host contamination issue

The audit build uses swig to generate a python wrapper.
Unfortunately, the swig info file references host include
directories.  Some of these were previously noticed and
eliminated, but the one fixed here was not.

Upstream Status:  pending

Signed-off-by: Anders Hedlund <anders.hedlund@windriver.com>
Signed-off-by: Joe Slater <jslater@windriver.com>
---
 bindings/swig/python/Makefile.am | 3 ++-
 bindings/swig/src/auditswig.i    | 4 ++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/bindings/swig/python/Makefile.am b/bindings/swig/python/Makefile.am
index ae7c52b..d1bb93c 100644
--- a/bindings/swig/python/Makefile.am
+++ b/bindings/swig/python/Makefile.am
@@ -22,6 +22,7 @@
 CONFIG_CLEAN_FILES = *.loT *.rej *.orig
 AM_CFLAGS = -fPIC -DPIC -fno-strict-aliasing
 PYINC ?= /usr/include/$(PYLIBVER)
+STDINC ?= /usr/include
 AM_CPPFLAGS = -I. -I$(top_builddir) -I${top_srcdir}/lib -I${PYINC}
 SWIG_FLAGS = -python
 SWIG_INCLUDES = -I. -I$(top_builddir) -I${top_srcdir}/lib -I${PYINC}
@@ -35,7 +36,7 @@ _audit_la_DEPENDENCIES =${top_srcdir}/lib/libaudit.h ${top_builddir}/lib/libaudi
 _audit_la_LIBADD = $(top_builddir)/lib/libaudit.la
 nodist__audit_la_SOURCES  = audit_wrap.c
 audit.py audit_wrap.c: ${srcdir}/../src/auditswig.i 
-	swig -o audit_wrap.c ${SWIG_FLAGS} ${SWIG_INCLUDES} ${srcdir}/../src/auditswig.i 
+	swig -o audit_wrap.c ${SWIG_FLAGS} ${SWIG_INCLUDES} -I$(STDINC) ${srcdir}/../src/auditswig.i 
 
 CLEANFILES = audit.py* audit_wrap.c *~
 
diff --git a/bindings/swig/src/auditswig.i b/bindings/swig/src/auditswig.i
index 9364ac4..48667d4 100644
--- a/bindings/swig/src/auditswig.i
+++ b/bindings/swig/src/auditswig.i
@@ -39,8 +39,8 @@ signed
 #define __attribute(X) /*nothing*/
 typedef unsigned __u32;
 typedef unsigned uid_t;
-%include "/usr/include/linux/audit.h"
+%include "linux/audit.h"
 #define __extension__ /*nothing*/
-%include "/usr/include/stdint.h"
+%include "stdint.h"
 %include "../lib/libaudit.h"
 
-- 
1.8.4.2

